src/RevCPU.cc

```diff
@@ -676,57 +660,39 @@ bool RevCPU::clockTick( SST::Cycle_t currentCycle ){
 
 // Initializes a RevThread object.
 // - Moves it to the 'Threads' map
-// - Adds it's ThreadID to the ThreadQueue to be scheduled
-void RevCPU::InitThread(std::shared_ptr<RevThread>& ThreadToInit){
-
+// - Adds it's ThreadID to the ReadyThreads to be scheduled
+void RevCPU::InitThread(std::unique_ptr<RevThread> ThreadToInit){
+  // @Lee, can this be a reference? I'm not looking to transfer ownership of the regfile pointer
+  std::unique_ptr<RevRegFile> RegState = ThreadToInit->TransferVirtRegState();
   auto gp = Loader->GetSymbolAddr("__global_pointer$");
-  ThreadToInit->GetRegFile()->SetX(RevReg::gp, gp);
-  ThreadToInit->GetRegFile()->SetX(RevReg::s0, gp);
-
-  uint32_t TID = ThreadToInit->GetThreadID();
-  // Check if this ThreadID has already been assigned... if so... something has gone horribly wrong
-  // print out all threads
-  auto it = Threads.find(TID);
-  if( it != Threads.end() && it->second->GetState() != ThreadState::START ){
-    output.fatal(CALL_INFO, 99, "Error: ThreadID %" PRIu32 " has already been assigned... this is a bug.\n", TID);
-  }
-  output.verbose(CALL_INFO, 4, 0, "Initializing Thread %" PRIu32 "\n", TID);
+  RegState->SetX(RevReg::gp, gp);
+  RegState->SetX(RevReg::s0, gp); // This can be used as the fp reg per ABI
```

Comment:
In your `enum class` is it possible to have an `fp` register point to the same value as the `s0` register? Or is it a single key only

Reply:
> In your `enum class` is it possible to have an `fp` register point to the same value as the `s0` register? Or is it a single key only

Already `fp` registers and `x` (gpr) registers have tags which are the same values. What `fp` or `gpr` register do you want added/changed?

The names of the registers are their ABI names explicitly specified in the RISC-V Architecture SPEC (like `a0`, `sp`, `gp`, `ra`). This is intentional because those names are more familiar. Is there a name you need which is not a part of the ABI naming convention?

You can use integers instead of `RevReg` enums to give a raw register number, which is how `rs1`, `rs2`, `rs3` and `rd` are handled. The functions taking registers are templated to accept either integers or `RevReg` and are explicitly cast to `size_t`  before using them as a raw register number.

Reply:
> In your `enum class` is it possible to have an `fp` register point to the same value as the `s0` register? Or is it a single key only

Sorry I misunderstood. `fp` is the official ABI name of frame pointer, which is why it's used in the `RevReg` enum.

I was referring to your **comment** after `//`. "Frame pointer" should be used in the comment instead of `fp` which is ambiguous and means "floating point" much more often.

Reply:
"This can be used as the fp reg..." should be "This can be used as the frame pointer reg"

Reply:
"fp reg" 99% of the time means "floating point register", so "frame pointer" should be worded explicitly to avoid confusion.

`RevReg::fp` is the ABI's official name for the frame pointer register. The `RevReg::` prefix makes that unambiguous. Bare "fp" used in comments almost always means floating-point.

Reply:
... and since there are "fp registers", "the fp register" is even more ambiguous.

Reply:
Ah yeah those are all very valid points